### PR TITLE
Feature piece triggers

### PR DESCRIPTION
**Pull request by abuaboud**

### CODE REVIEW

File: src/worker/packages/backend/src/instance/instance-side-effects.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            return;
+        }
+
+        const flowVersionIds = Object.values(instance.flowIdToVersionId);
+
+        const flowVersions = await flowVersionRepo.findBy({
+            id: In(flowVersionIds),
+        });
+
+        const disableTriggers = flowVersions.map(triggerUtils.enable);
```
Comment (abuaboud) [inline src/worker/packages/backend/src/instance/instance-side-effects.ts]: I think this is need to be disable

File: src/worker/packages/backend/src/instance/instance-side-effects.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        await Promise.all(disableTriggers);
+    }
+};
+
+const lockFlowVersions = async (flowVersions: FlowVersion[]): Promise<void> => {
```
Comment (abuaboud) [inline src/worker/packages/backend/src/instance/instance-side-effects.ts]: can we please lock collection version as well

Comment (khaledmashaly) [reply src/worker/packages/backend/src/instance/instance-side-effects.ts]: yes see 6e0b91b

File: src/worker/packages/backend/src/instance/instance-side-effects.ts (hunk 1 above)
Comment (khaledmashaly) [reply src/worker/packages/backend/src/instance/instance-side-effects.ts]: oops good catch see 4f45c19

**Comment by john-jova** [comment]
Phase 3 complete — merged to main. All VM tests pass.

**Pull request merged**